mmv1/products/dlp/DeidentifyTemplate.yaml

```diff
@@ -42,8 +42,17 @@ examples:
     primary_resource_id: 'basic'
     test_env_vars:
       project: :PROJECT_NAME
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'dlp_deidentify_template_with_template_id'
+    primary_resource_id: 'with_template_id'
+    vars:
+      name: 'my-template'
+    test_env_vars:
+      project: :PROJECT_NAME
 custom_code: !ruby/object:Provider::Terraform::CustomCode
-  encoder: templates/terraform/encoders/wrap_object.go.erb
+  decoder: templates/terraform/decoders/dlp_template_id.go.erb
+  encoder: templates/terraform/encoders/wrap_object_with_template_id.go.erb
+  update_encoder: templates/terraform/update_encoder/wrap_object.go.erb
```

Comment:
I think it might be easier to reuse `templates/terraform/encoders/wrap_object.go.erb`, instead of duplicating that functionality in the `update_encoder` directory. Would that work here? (Same for InspectTemplate)

And, just to verify, I assume that `wrap_object_with_template_id` doesn't work for this case? (presumably because `templateId` cannot be updated?)

Reply:
Yeah, Reusing the `templates/terraform/encoders/wrap_object.go.erb` would definitely work. Also, while updating the resource, the [request body](https://cloud.google.com/dlp/docs/reference/rest/v2/projects.deidentifyTemplates/patch#request-body) doesn't expect `templateId` field. Hence, I've used two different encoders for creating and updating the template.